pkg/operator/controller/ingress/status.go

```diff
@@ -822,25 +819,37 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 			},
 		}
 	}
-
-	if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) {
-		return []operatorv1.OperatorCondition{
+	var conditions []operatorv1.OperatorCondition
+	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS ||
+		(ic.Spec.EndpointPublishingStrategy != nil && ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+			ic.Spec.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS &&
```

Comment:
What situation would `ic.Spec` be important when `ic.Status` don't have it set? Is this just being extra cautious?

Reply:
So this is just me being over cautious in the scenario where `dnsManagementPolicy` isn't set on the spec (relying on the defaults, which later gets synced to `ic.Status`) and we migrate to `Unmanaged` by editing the `spec`. (if `spec` didn't sync to `status`).

I've removed it, on hindsight this makes it less readable.

Reply:
Yea I don't have strong feelings either way. If `ic.Status` isn't eventually getting updated, then I suppose something went wrong, and DNS could be considered still managed. There might be more nuance to these situations than I am aware up. 

But I'm good with just using `ic.Status`.